molgenis-core-ui/src/main/resources/js/component/Table.js

```diff
@@ -221,6 +234,9 @@
 		},
 		_handlePageChange: function(e) {
 			this._refreshData(this.props, _.extend({}, this.state, {start: e.start}));
+		},
+		_handleRowsPerPageChange: function(e) {
+			this.setProps({maxRows: e.target.value});
```

Comment:
maxRows should be moved to component state (see https://facebook.github.io/react/docs/interactivity-and-dynamic-uis.html#what-should-go-in-state). Transfer the props passed to the component on creation to state.